atom_tools/lib/regex_utils.py

```diff
@@ -71,6 +72,11 @@ class FilteringPatternCollection:
         '{objectSlices: objectSlices[?ATTRIBUTECONDITION`TARGET_VALUE`], '
         'userDefinedTypes: userDefinedTypes[?ATTRIBUTECONDITION`TARGET_VALUE`]}'
     )
+    jmespath_purls = jmespath.compile('reachables[].purls[]')
+    purl_pkg = re.compile(r'(?P<p1>[^/:]+/(?P<p2>[^/]+))(?:(?:.|/)v\d+)?(?=@)')
```

Comment:
It's not as simple as this since there are language specific hacks. Best to use the official python library